demo/core_13_$.utils.code.js

```diff
@@ -136,3 +136,60 @@ $.utils.code.toSourceSafe = function(value) {
     throw e;
   }
 };
+
+$.utils.code.formatForEval = function(src) {
+  // Eval treats {} as an empty block (return value undefined).
+  // Eval treats {'a': 1} as a syntax error.
+  // Eval treats {a: 1} as block with a labeled statement (return value 1).
+  // Detect these cases and enclose in parenthesis.
+  // But don't mess with: {var x = 1; x + x;}
+  // This is consistent with the console on Chrome and Node.
+  try {
+    var ast = $.utils.acorn.parse(src, { ecmaVersion: 5 });
+  } catch (e) {
+    try {
+      ast = $.utils.acorn.parseExpressionAt(src, 0, { ecmaVersion: 5 });
+      if (src.substring(ast.end).trim() !== '') {
+        throw SyntaxError('Not just an expression');
+      }
+    } catch (e) {
+      // This eval will fail.  Have fun.
+      return src;
+    }
+    // This is an object literal: {'a': 1}
+    return '(' + src + ')';
+  }
+  if (ast.type === 'Program' && ast.body.length === 1 &&
+      ast.body[0].type === 'BlockStatement') {
+    if (ast.body[0].body.length === 0) {
+      // This is an empty object: {}
+      return '(' + src + ')';
+    }
+    if (ast.body[0].body.length === 1 &&
+        ast.body[0].body[0].type === 'LabeledStatement' &&
+        ast.body[0].body[0].body.type === 'ExpressionStatement') {
+      // This is an object literal: {a: 1}
+      return '(' + src + ')';
+    }
+  }
+  return src;
+};
+
+$.utils.code.formatForEval.test = function() {
+  var matrix = {
+    '1 + 2': '1 + 2',
+    '1 + 2;\n3 + 4;': '1 + 2;\n3 + 4;',
+    '{}': '({})',
+    '{};\n3 + 4;': '{};\n3 + 4;',
+    '{"a": 1}': '({"a": 1})',
+    '{"a": 1};\n3 + 4;': '{"a": 1};\n3 + 4;',
+    '{a: 1}': '({a: 1})',
+    '{a: 1};\n3 + 4;': '{a: 1};\n3 + 4;'
+  };
+  for (var key in matrix) {
```

Comment:
Add `hasOwnProperty` check here.  After all, in-core `Object.prototype` very likely to accidentally get enumerable properties on occasion; don't want tests breaking (unless they are specifically testing that, of course).

Reply:
Done.

Reply:
I think it would be preferable to dispose of the unexpected case at the top and avoid indenting the main body of the loop more than necessary:

```JS
  for ( ... ) {
    if (!cases.hasOwnProperty(key)) continue;
    var actual = ...
```

Reply:
Done